src/components/MenuLink/index.tsx

```diff
@@ -35,7 +21,28 @@ MenuLink.propTypes = {
    * - `import { Link } from 'react-router'`
    * - `Link={Link}`
    */
-  Link: PropTypes.elementType,
-};
+  Link?: any;
```

Comment:
<img width="601" alt="Screen Shot 2021-12-14 at 7 46 13 AM" src="https://user-images.githubusercontent.com/708820/146010350-2b9b21df-44ff-4dc4-b8ac-d0b00c2ad4e7.png">

I don't necessarily want you to do this change (at least right now, we should discuss this later) but this is one way to type this.

What I'm doing here is grabbing the prop interface of `ReachMenuLink` and then I'm accessing the typedef of `as` directly. This works.

I don't necessarily want to do this, as we have many `Link` props scattered around, and I don't want us just pulling on it in one place. Ideally we need to figure out how to type `Link` ourselves and then make sure it can be passed into theirs. But I just wanted to show you this technique. I learned about most of this yesterday, and figured out how to simplify it through executeprogram.

Reply:
this is interesting.
I agree about our types, we need to figure out a way to have them all placed correctly, I will add a `TODO` so we don't lose it.